include/ttmlir/Dialect/TTIR/Pipelines/TTIRPipelines.h

```diff
@@ -0,0 +1,31 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef TTMLIR_DIALECT_TTIR_PIPELINES_PASSES_H
+#define TTMLIR_DIALECT_TTIR_PIPELINES_PASSES_H
+
+#include "mlir/Pass/PassOptions.h"
+
+namespace mlir::tt::ttir {
+
+// Options for the TTIR to TTNN backend pipeline.
+//
+struct StableHLOToTTIRPipelineOptions
+    : public PassPipelineOptions<StableHLOToTTIRPipelineOptions> {
+  // Option to enable --remove-dead-values optimization pass.
+  Option<bool> removeDeadValuesEnabled{
```

Comment:
@svuckovicTT, can you take a look at this? We wanted to do this some time ago, but we found a way to remove the ops during the pass's execution, right?

Reply:
I can also remove the op during pass execution, that was the first solution I had in mind, but then I realized that this is gonna require much more complex analysis. If you take a look at example I listed [here](https://github.com/tenstorrent/tt-mlir/pull/585#discussion_r1741263000) there is a constant OP which is used in reduce OP afterwards. I can just remove the constant OP, but then I would have to check not just the reduce OP, but also all the other OPs in that block to see if they use the operand that constant OP produced. That looked like some optimization that MLIR should already have, and indeed I found the `--remove-dead-values` pass.

Reply:
Yeah, I see your point, and I agree with you.